### PR TITLE
added example sinus_fit

### PR DESCRIPTION
**Pull request by edeforas**
This example is a simple function fit using DNN (slighlty more complex than a xor however).

### CODE REVIEW

File: examples/sinus_fit/sinus_fit.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/*
+    Copyright (c) 2018, Etienne de Foras and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
```
Comment (beru) [inline examples/sinus_fit/sinus_fit.cpp]: @edeforas Even though you are the one who added this file, I prefer to see uniformed file header copyright.
```
/*
    Copyright (c) 2013, Taiga Nomi and the respective contributors
    All rights reserved.
    Use of this source code is governed by a BSD-style license that can be found
    in the LICENSE file.
*/
```
My concern is, if each of the contributors used different copyright notices, that might lead to confusions.

By the way, `AUTHORS` file can be updated by script. Please see https://github.com/tiny-dnn/tiny-dnn/issues/228#issuecomment-233293328

File: examples/sinus_fit/sinus_fit.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+// this example show how to use tiny-dnn library to fit data, by learning a
+// sinus function
+
+// please see also at:
```
Comment (beru) [inline examples/sinus_fit/sinus_fit.cpp]: `please also see:` should be enough.

File: examples/sinus_fit/sinus_fit.cpp [hunk 3]
```diff
@@ -0,0 +1,85 @@
+/*
+    Copyright (c) 2018, Etienne de Foras and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+
+// this example show how to use tiny-dnn library to fit data, by learning a
```
Comment (beru) [inline examples/sinus_fit/sinus_fit.cpp]: Plural verb should be used here, `this examples shows`.

File: examples/sinus_fit/sinus_fit.cpp [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+// sinus function
```
Comment (beru) [inline examples/sinus_fit/sinus_fit.cpp]: I've heard adding a period at the end of sentence is preferable or Brits will kidnap you and you'll end up having their food for the rest of your entire life.

File: examples/CMakeLists.txt [hunk 5]
```diff
@@ -63,7 +63,11 @@ add_executable(example_test_char_rnn char_rnn/test_char_rnn.cpp ${tiny_dnn_heade
 target_link_libraries(example_test_char_rnn
         ${project_library_target_name} ${REQUIRED_LIBRARIES})
 
-cotire(example_deconv_visual example_cifar_train example_cifar_test example_recurrent_addition example_train_char_rnn example_test_char_rnn)
+add_executable(example_sinus_fit sinus_fit/sinus_fit.cpp ${tiny_dnn_headers})
+target_link_libraries(example_sinus_fit
+        ${project_library_target_name} ${REQUIRED_LIBRARIES})
+
+cotire(example_deconv_visual example_cifar_train example_cifar_test example_recurrent_addition example_train_char_rnn example_test_char_rnn sinus_fit)
```
Comment (beru) [inline examples/CMakeLists.txt]: I see `sinus_fit` at the end of `cotire` command's argument list. Shouldn't that be `example_sinus_fit`?

**Comment by edeforas** [comment]
Fixes commited, thanks for your code review!

**Pull request merged**